mineral-ui/mineral-ui
# [Website] Add home page; rework interior pages

### Description
* Add `siteColors` and `siteTheme`, which includes baseline units and shared breakpoints
* Add other themes specific to app portions, e.g. the home page, the hero on the home page, and the interior nav
* Add TypeKit fonts
* Add logo assets, including favicons
* Update prism theme
* Refactor routing in App and Router to simplify and consolidate
* `?chromeless` now only works when viewing a single component example
* Refactor usage of Helmet to simplify and consolidate
* Add `anchors` prop to Markdown to disabled heading anchor links
* Rework and add some site-specific versions of components to avoid theming conflicts
    * Link -> SiteLink
    * Heading -> Heading & SiteHeading
    * Add SiteButton
* Fully-responsive, with a sticky side nav and component page sub nav
* Add home page, with animating triangle backgrounds and a theme playground
* GuidelinePage -> Page, which is basically an app shell for our site
    * Pages in different sections use a different theme
* Remove titles from page markdown content, in favor of displaying the title from the page object
* Ensure live previews and Best Practice examples render with proper themes
* Refactor ComponentDoc page and redesign Best Practices section
* Styling
    * Headings and a few other bits use TypeKit fonts
    * Text content is (mostly) aligned to the baseline (append `?grid` to the URL)
    * Many style refactors, improvements, and tweaks throughout
* Refactor Button demo examples to use a common `DemoLayout`
* Update snapshots

#### Notes
* The per-pages section colors are not final

### Motivation and context

We need a home page and interior pages need to match it.

Closes #226, closes #425, closes #430

### Screenshots, videos, or demo, if appropriate


https://226-home-page--mineral-ui.netlify.com/

#### Home, _before_
[this area intentionally left blank]

#### Home, _after_
![New home page screenshot](https://user-images.githubusercontent.com/486540/32580542-fea3ca4c-c4a2-11e7-9618-f942b7ee4b1b.png)

#### Component page, _before_
![Old component page screenshot](https://user-images.githubusercontent.com/486540/32580549-09d33ede-c4a3-11e7-8fee-cfa2abfa911b.png)

#### Component page, _after_
![New component page screenshot](https://user-images.githubusercontent.com/486540/32580561-161a489a-c4a3-11e7-9028-c1b9983870c3.png)

### How to test


1. `cd mineral-ui && npm start`
2. Notice that you're on the brand-new home page
3. Check all pages (including an individual component example, chromeless and not) for anything that looks amiss
4. Specifically confirm that all example live previews and Best Practice examples look exactly the same as before
5. Check all screen sizes

### Types of changes

- Other (provide details below)

This is almost all website-only. The small changes to styles and themes have no impact on the components, so they're classified as a `refactor`.

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support) **[1]**
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered **[2]**



**[1]** - We intentionally used some fancy CSS Grid and `position: sticky` goodies that have appropriate fallbacks for non-supporting browsers. Other than a hilariously ugly, but fully-functional theme playground on the home page in IE11, things look surprisingly good.
**[2]** - The animating triangles burn up the CPU. They will be replaced with a much more respectable "static" (but still animated) SVG in a future PR.

### How does this PR make you feel?

![home](https://media.giphy.com/media/5xtDarqCp0eomZaFJW8/giphy.gif)
